### DISCUSSION TITLE
How to adjust the time step during calculation?

### DISCUSSION THREAD

**Discussion by LiuTingy** [Anura3D/Anura3D_OpenSource#33] [General]
Hello everyone:
I am using Anura3D 2022 version. There exists some numerical oscillation in my calculation. The incipient calculation results are in line with expectations, but distortion occurs when the deformation reaches a certain degree. I'm wondering if there is an option to adjust the critical timestep during dynamic analysis. I don't see any tips in the manual tutorial.
I hope someone can help me to solve the problem, thank you very much!

**Comment by LiuTingy** [comment]
Maybe I can reduce the courant number to reduce the time step, so as to reduce the numerical oscillation, but I don't know whether it‘s feasible for dynamic analysis and what's the appropriate range taking into account of the calculation efficiency and accuracy.

**Reply by ayerrocolom** [reply]
You can indeed modify the time step by decreasing the Courant number, which is a factor that multiplied the "critical time step" calculated intrinsically in Anura3D based on established criteria (e.g., CFL condition). The size of the time step will not directly affect the dynamics of the problem. Usually, a time step with a Courant number slightly less than 1 is enough for solution convergence. However, in some cases, when the problem is more complicated (e.g., tricky meshes, complex geometries, very different materials, contact algorithms, multi-phase formulations.... all together), the Courant number needs to be decreased. There is no "good"range, it depends on the problem, and the user should assess that based on solution convergence and calculation efficiency.

**Reply by LiuTingy** [reply]
Thank you so much for the explanation. I will try to find an appropriate number for my model. And I guess if we can plot the courant number field, or will you guys include it in the next development plan? Because in some CFD simulations, users can visually observe the distribution area and size of the number, thus facilitating the personalized adjustment of their models.

**Reply by ayerrocolom** [reply]
The Courant number is selected by the user, not by the code, and it is a constant number across the mesh and during the full calculation.

**Comment by Armin-Geranghadr** [comment]
Hello everyone,

I have a technical issue in adjusting time steps. As you can see in screen, I inserted 50 number of calculation steps and 0.05s time per calculation step. However, after the calculation I got this warning that my time step is 1 and I have only access to time 0 and 1 in my results. 
Could anyone help me?

![1](https://github.com/user-attachments/assets/7f45d0fa-c602-4f8f-b7fb-cefaaf49afe6)
![2](https://github.com/user-attachments/assets/d7eba43c-e94c-42b1-95c5-15a3d508b0ba)

**Reply by pmarveggio** [reply]
It seems the simulation is not completed yet. You may have to wait a bit more. Try to check with smaller time steps and see if the analysis reaches the end of calculation.

**Reply by HaoranzzzZZZ** [reply]
Hi,
Have you solved this issue? I'm trying to do the one-dimensional consolidation case according to the tutorial and I have the same problem.  
<img width="480" height="300" alt="calculation" src="https://github.com/user-attachments/assets/18fe19f6-c18c-4dbe-ac2e-1d02df330b36" />
It shows that the simulation is finished, but there is only the result for 1 step.
<img width="423" height="300" alt="result_consolidation" src="https://github.com/user-attachments/assets/1d3b318a-904b-4fd8-b034-6d3ff6cdb5b1" />

**Reply by AgustinToroF** [reply]
Hi @HaoranzzzZZZ. Do you have a material point outside the soil column? In that case, I recommend you select material points inside the column of soil and later calculate.

**Reply by HaoranzzzZZZ** [reply]
Hi @AgustinToroF, thank you for your reply. 
<img width="98" height="677" alt="1" src="https://github.com/user-attachments/assets/c525fa9a-9aa3-4260-a862-b02987013ec6" /><img width="117" height="679" alt="2" src="https://github.com/user-attachments/assets/56dc62cd-ecc4-44c9-834e-ea910ffb82b3" /><img width="383" height="683" alt="3" src="https://github.com/user-attachments/assets/034f015d-ab19-4f6f-b1b1-09f3b3a15866" />
I've checked the initial and final state, no material point is outside the soil solumn. The timestep settings are also attached here.

**Reply by AgustinToroF** [reply]
In your last figure, draw values of "group: Select elements for MPs output Auto1" and verify that, the material points to output information are inside the soil column.

**Reply by HaoranzzzZZZ** [reply]
> In your last figure, draw values of "group: Select elements for MPs output Auto1" and verify that, the material points to output information are inside the soil column.

Sorry, as a newbie I didn't find a way to display the group, could you please help me to see what's wrong with the case setup, thanks!
[Test_consolidation.gid.zip](https://github.com/user-attachments/files/21499832/Test_consolidation.gid.zip)
[Test_consolidation.A3D.zip](https://github.com/user-attachments/files/21499837/Test_consolidation.A3D.zip)

**Reply by AgustinToroF** [reply]
I checked the "Test_consolidation.gid" and keep thinking that the problem is that your seleted a element's material point which doesn't belong inside the soil column. To verify that, just right click on "group: Select elements for MPs output Auto1", then click on "Draw" and finally on "Draw values". You'll see that one material point belongs an void element. Therefore, select an element for MPs output inside the soil column, eg. the top soil column and calculate.

I corrected the issue and Anura3D calculated successfull the oedometer test. Tell me if you were able to do it.

Nobody is born knowing and I had the same problem months ago.

**Reply by HaoranzzzZZZ** [reply]
> I checked the "Test_consolidation.gid" and keep thinking that the problem is that your seleted a element's material point which doesn't belong inside the soil column. To verify that, just right click on "group: Select elements for MPs output Auto1", then click on "Draw" and finally on "Draw values". You'll see that one material point belongs an void element. Therefore, select an element for MPs output inside the soil column, eg. the top soil column and calculate.
> 
> I corrected the issue and Anura3D calculated successfull the oedometer test. Tell me if you were able to do it.
> 
> Nobody is born knowing and I had the same problem months ago.

At first I misunderstood what you meant and thought it was to show the MPs group in paraview.I followed your suggestion and select other elements for MPs output inside the soil column, the case runs fine now. 
Thank you so much!

**Comment by lvtorio-dev** [comment]
I would like to seek help on how to show the results for each time steps (0.05s, 0.1s, 0.2s,..., 2.5s)? This is my first time using paraview and it just shows time 0 and 1 in the result. I hope someone will be able to help. Thank you!
<img width="1287" height="1600" alt="Gid" src="https://github.com/user-attachments/assets/618a8e9a-19eb-42db-a337-325b8f45b893" />
<img width="2560" height="1600" alt="paraview" src="https://github.com/user-attachments/assets/dd848b38-e139-474a-94dd-76c6981501b5" />

**Reply by AgustinToroF** [reply]
Did you follow the instructions in the 2025 Tutorial Manual? I ask this because the manual selects elements for MP output outside the soil skeleton, which doesn't make much sense since there are no material points initially associated with it outside the soil.
